### ISSUE TITLE
Consider centralizing our assets and style declarations

### ISSUE THREAD

**Issue by radotreyes** [austintackaberry/ydkjs-exercises#194]
We're using styled components to handle the majority of our CSS right now. Currently, our file structure for our styled components looks like this:

![image](https://user-images.githubusercontent.com/25578179/45592029-6da95380-b917-11e8-8af7-08cb35eeaf7d.png)

You can see that each component has its own `styled.js` module which is being imported inside the corresponding component. This is a solid idea, as it's clear which component is using which styles, however because it turns out we are reusing many of our styles, it does pose a few problems:

1. There are instances where things like buttons, headers, etc. are defined multiple times across different components. For example, `Question` and `Sidebar` both have their own `FlatButton` styled components, and they are almost identical.

2. Say for example: We're in `BookHome` and we want to use a style from `ChapterHome`: We would be importing a style from `ChapterHome` and using it in `BookHome`, which I don't think is very clear or intuitive.

3. Making changes to a single style, for example `FlatButton`, might also call for us to make changes to every other `FlatButton` in every other component which implements it. This isn't good!

So, I propose that we centralize our assets/styles into one folder, maybe simply named `assets` or `styles`. I do think that it would resolve a lot of what I explained above.

Let me know what you think!

**Comment by austintackaberry** [comment]
Does that mean that we would have all of our styles in one location? I like the idea of having styles scoped to a component. Could we have, for example, a common button stored somewhere and then if a styled-component needed to use it then it just imported it?

**Comment by radotreyes** [comment]
Yeah, that's a good solution! The main issue I see here is when common styles/assets have to be reused, they're being imported from another component's directory. So I think for all of our _common_ assets, a corresponding folder would be good.

Anything else that's isolated to one component should stay where it is. 🙂

**Comment by designviacode** [comment]
@radotreyes @austintackaberry 
I'd like to help with this one.

couple of quick suggestions:
- folder could be named **`shared`**, either inside the `src` or `src/components`
- could have sub-directories inside the `shared` directory, such as:
  - `basic`
  -  `compound`
  - `hoc` ( can help with later-addition as well )

**Comment by austintackaberry** [comment]
@designviacode Can you explain what would be in each of those folders? (basic, compound, hoc)

**Comment by radotreyes** [comment]
@designviacode I like the idea of your `basic` folder. I think it would be useful to store our [themes](https://www.styled-components.com/docs/advanced) and other common components in there. 

I think the idea of a `compound` folder is sound, and I do agree that we should have a system for dealing with derived/child components. I don't really think that another subdirectory is necessary for that though; I'd rather handle that with styled-components' [`css` function](https://www.styled-components.com/docs/api#css).

New contributors who are writing new styled components might see the `basic` and `compound` folders and not know which one to use.  That might also be risking architectural debt because someone will inevitably put a component in a wrong folder.

I think one `basic` or `shared` (`styled`?) folder would be good:

structure:
```
src/
└── styled/
    ├── Button/
 	|   ├── index.js
    ├── ResetButton/
 	|   ├── index.js
    ├── Modal/
    |   └── index.js
    └── index.js
```

`src/styled/index.js`:
```
import Button from './Button'
import ResetButton from './ResetButton'

/* etc. */
```
